### PR TITLE
Fix setting default_ttl to 0 on create

### PR DESCRIPTION
**Pull request by spiralman**
This change allows a service to be created with a  `default_ttl` of `0`, fixing #204.

The problem was that, for integer values, the "old" value in the `ResourceData` is `0`, meaning that, if the "new" value is `0`, `HasChange(key)` returns `false`.

It didn't seem possible to override this behavior, either by modifying the "old" value in the `ResourceData` or by specifying a `CustomizeDiff` function (which appears to only be able to *remove* differences, not add ones that wouldn't, otherwise, be detected). Note: if there *is* a way to accomplish this, and that is the preferred method to fix these issues, let us know and we can update the PR.

Instead, we have added an explicit check for the `default_ttl` being set to `0` on service creation and we force the update call to be made, in that case.

We aren't very familiar with the tools available inside the Terraform Plugin SDK, so we may have missed a better solution. Please let us know if there's a better way to fix this issue.

### CODE REVIEW

**Comment by spiralman** [comment]
@philippschulte Not sure what the procedure is for requesting code review in this project, but it seems like you are the primary maintainer. Do you think we could get this into the next release, if you're OK with the solution?

File: fastly/resource_fastly_service_v1.go
Comment (phamann) [inline fastly/resource_fastly_service_v1.go:1629]: Can you please add a comment above this explaining why the third conditional exists.

**Review by phamann — CHANGES_REQUESTED** [review]
:+1: Thank you for your contribution @spiralman I'm happy with the implementation. Only one minor request around adding a comment to explain why the logic exists.

**Comment by spiralman** [comment]
@phamann Thanks for the review! Updated the PR with a code comment explaining why the check is necessary. Thanks!

**Review by phamann — APPROVED** [review]
👍 LGTM

**Comment by spiralman** [comment]
@phamann Thanks for the review! Is there anything else necessary to get this merged and into a release? 

Currently, this bug is hindering our ability to roll out some new services in Fastly. We'd rather not have to build some sort of automation on top of Terraform (i.e. run `apply` twice) to work around the bug, if we can get a fix out relatively soon.

Please let me know if there's anything else we can do to help.

**Comment by phamann** [comment]
@spiralman apologies, I also wanted to confirm @philippschulte is happy with this before merging.

**Review by philippschulte — APPROVED** [review]
@spiralman the changes look good to me! Thanks for also including the tests! I'll reach out to Hashicorp to request a new release.

**Comment by spiralman** [comment]
Great, thanks very much @philippschulte and @phamann!

**Pull request merged**
